WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java

```
@@ -98,10 +101,15 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         String requestedDate = intent.getStringExtra(ARG_DATE);
         if (requestedDate == null) {
             AppLog.w(T.STATS, "StatsService is started with a NULL date on this blogID - "
-                    + mServiceBlogId + ". Using current date!!!");
-            int localTableBlogId = WordPress.wpDB.getLocalTableBlogIdForRemoteBlogId(
-                    Integer.parseInt(mServiceBlogId));
-            requestedDate = StatsUtils.getCurrentDateTZ(localTableBlogId);
+                    + blogId + ". Using current date!!!");
+            try {
+                int parsedBlogID = Integer.parseInt(blogId);
+                int localTableBlogId = WordPress.wpDB.getLocalTableBlogIdForRemoteBlogId(parsedBlogID);
+                requestedDate = StatsUtils.getCurrentDateTZ(localTableBlogId);
+            } catch (NumberFormatException e) {
```

Probably not necessary (I fear that could hide other errors).